modules/ding_nodelist/js/node_blocks.js

```diff
@@ -58,6 +60,71 @@
 
         $(this).find('.title-and-lead').css('min-height', title_and_lead_height);
       });
+
+      /**
+       * Node Blocks DOM manipulation.
+       */
+      var nbPanes = $('.ding_nodelist-node_blocks .ding_nodelist-items', context);
+      nbPanes.each(function (key, pane) {
+        var $pane = $(pane);
+
+        // Build default config parameters.
+        var config = [];
+        $pane.find('article').each(function (i, val) {
+          var data = {};
+          data.id = $(val).data('id');
+          data.hasImage = $(val).hasClass('has-image');
+          data.marginTop = 0;
+
+          if ($(val).hasClass('has-image') && i !== 0 || !$(val).hasClass('has-image') && i !== 0) {
+            data.marginTop = 200;
+          }
+          config.push(data);
+        });
+
+        // Add corrections to the configs conditionally.
+        config.map(function (val) {
+          var current = config[val.id];
+          var next = config[val.id + 1];
+
+          if (next === undefined) {
+            return;
+          }
+
+          if (current.hasImage === true && next.hasImage === false) {
+            next.marginTop = 200;
+          }
+          else if (current.hasImage === false && next.hasImage === true) {
+            next.marginTop = 0;
+          }
+          else if (current.hasImage === false && next.hasImage === false) {
+            next.marginTop = 0;
+          }
+        });
+
+        // Process items rendering.
+        config.forEach(function (e) {
+          if (isMobile) {
+            if (nodeBlocksSettings.displayMobileImage === 1) {
+              if (e.hasImage === true) {
+                $pane.find("article[data-id=" + e.id + "]").find('.nb-image').css('display', 'block');
+
+                $pane.find("article[data-id=" + e.id + "]").css('margin-top', e.marginTop);
+                $pane.find("article[data-id=" + e.id + "] .inner").css('margin-top', e.marginTop);
+                if (e.marginTop === 0) {
+                  $pane.find("article[data-id=" + e.id + "] .inner").css('margin-top', 200);
+                }
+                if (e.id === config.length - 1) {
+                  $pane.find("article[data-id=" + e.id + "]").css('margin-bottom', $pane.find("article[data-id=" + e.id + "]").height());
+                }
+              }
+              else {
+                $pane.find("article[data-id=" + e.id + "]").css('margin-top', e.marginTop);
+              }
+            }
+          }
+        });
+      });
```

Comment:
This looks badly convoluted. Please reduce the nesting and do not apply inline css. Apply classes instead.